app/common/auth/__init__.py

```diff
@@ -118,6 +127,8 @@ def sso_sign_in() -> ResponseReturnValue:
 def sso_get_token() -> ResponseReturnValue:
     result = build_msal_app().acquire_token_by_auth_code_flow(session.get("flow", {}), request.args)
     if "error" in result:
+        if MSAL_ERROR_AUTHORIZATION_CODE_WAS_ALREADY_REDEEMED in result.get("error_codes", []):
+            return redirect(url_for("auth.sign_out"))
```

Comment:
It might be nice to log warnings when this is happening so that we'd be able to dig in to if we're still getting this failure mode and users might be getting confused - as this will go from a very vocal 500 to something only experienced by users indirectly